server/testdriver/core/src/main/java/org/infinispan/server/test/core/ContainerInfinispanServerDriver.java

```diff
@@ -260,22 +304,36 @@ protected void start(String fqcn, File rootDir, File configurationFile) {
             });
       image.get();
       log.infof("Created image %s", name);
+   }
 
-      int numServers = configuration.numServers();
-      CountdownLatchLoggingConsumer clusterLatch = new CountdownLatchLoggingConsumer(numServers, String.format(CLUSTER_VIEW_REGEX, numServers));
-      if (configuration.isParallelStartup()) {
-         CountdownLatchLoggingConsumer startupLatch = new CountdownLatchLoggingConsumer(numServers, STARTUP_MESSAGE_REGEX);
-         IntStream.range(0, configuration.numServers()).forEach(i -> createContainer(i, startupLatch, clusterLatch));
+   /**
+    * Starts an additional server outside that isn't part of {@link InfinispanServerTestConfiguration#numServers()}
+    * number. This is useful to start servers at a later point.
+    * <p>
+    * This method can only be invoked after {@link #start(String)} has completed successfully
+    * @param expectedClusterSize The expected number of members in the cluster. Some implementations may not work if other nodes
+    *                            outside of this driver are clustered and the argument could be ignored.
+    * @param volumeName The name of the shared volume to be used when a node restarts
+    */
+   public void startAdditionalServer(int expectedClusterSize, String volumeName) {
+      try {
+         CountdownLatchLoggingConsumer clusterLatch = new CountdownLatchLoggingConsumer(1, String.format(CLUSTER_VIEW_REGEX, expectedClusterSize));
+         CountdownLatchLoggingConsumer startupLatch = new CountdownLatchLoggingConsumer(1, STARTUP_MESSAGE_REGEX);
+
+         log.infof("Starting new single server for container %d with volume %s", name, volumeName);
+         createContainer(createdContainers(), volumeName, startupLatch, clusterLatch);
          Exceptions.unchecked(() -> startupLatch.await(TIMEOUT_SECONDS, TimeUnit.SECONDS));
-      } else {
-         for (int i = 0; i < configuration.numServers(); i++) {
-            CountdownLatchLoggingConsumer startupLatch = new CountdownLatchLoggingConsumer(1, STARTUP_MESSAGE_REGEX);
-            createContainer(i, startupLatch, clusterLatch);
-            Exceptions.unchecked(() -> startupLatch.await(TIMEOUT_SECONDS, TimeUnit.SECONDS));
-         }
+         Exceptions.unchecked(() -> clusterLatch.await(TIMEOUT_SECONDS, TimeUnit.SECONDS));
```

Comment:
This was here before, but on the rolling upgrade context, we'll need to emphasize in the docs that the view must always be stable before proceeding with operations.

Reply:
I think that is implied already, but we can probably say when doing a user controlled rolling upgrade that should be the first step. K8s already requires this afaik.